ivadomed/loader/segmentation_pair.py

```diff
@@ -315,12 +315,12 @@ def convert_file_to_nifti(self, filename: str, extension: str, is_gt: bool = Fal
         """
         # For '.png', '.tif', '.tiff', '.jpg' and 'jpeg' extentions
         # Read image as 8 bit grayscale in numpy array (behavior TBD in ivadomed for RGB, RBGA or higher bit depth)
-        if "tif" in extension:
-            img = np.expand_dims(imageio.v2.imread(filename, format='tiff-pil'), axis=-1).astype(np.uint8)
-            if len(img.shape) > 3:
-                img = np.expand_dims(imageio.v2.imread(filename, format='tiff-pil', as_gray=True), axis=-1).astype(np.uint8)
-        else:
-            img = np.expand_dims(imageio.v2.imread(filename, as_gray=True), axis=-1).astype(np.uint8)
+        _img = imageio.v3.imread(filename)
+        
+        if len(_img.shape) < 3:
```

Comment:
```suggestion
        if len(_img.shape) > 3:
```

Could it be that the sign in this condition was reversed by accident?

Reply:
Ahh nope sorry about that this doesn't work.

Reply:
It's the "as_gray" conversion that's missing. It also changed name in v3 of imageio, can recall what it is but it warns about it @kanishk16 @hermancollin

Reply:
It should be `mode="L"` in the imread options, see my currently open ivadomed PR that was supposed to deal with it, before this PR was opened. https://github.com/ivadomed/ivadomed/pull/1292/files

Reply:
Yes. The problem is that `mode='L'` works fine for RGB/RGBA PNGs, but not for TIFFs. I wasn't able to find the proper way to do it with imageio.v3

Reply:
Hmm! Would it be bad practice to first convert the TIFFS to RGB/RGBA, then "load" them? Probably a waste of memory and time, I wonder if that could work somehow

Reply:
>Well currently the rgb/rgba TIFFs are the problem. I assume you mean converting the TIFFs to PNGs or something like this? It would probably work in that case.

Yes woops, mind blank on my part.

>Maybe we should open an issue on the imageio side to ask them the best practice. In the end, all we want to do is convert a multichannel TIFF image to grayscale, it shouldn't be that hard

Definitely we should do that - if it hasn't been done by someone else already! I think the only hiccup is I think tifs are dealt with another dependency, TIFFILE? So maybe there's a conflict between the functionality of those two packages when we want to convert to B&W?